manim/mobject/mobject.py

```diff
@@ -2600,25 +2643,31 @@ def init_size(num, alignments, sizes):
             buff_x = buff_y = buff
 
         # Initialize alignments correctly
-        def init_alignments(alignments, num, mapping, name, dir_):
+        def init_alignments(
+            alignments: str | None,
+            num: int,
+            mapping: dict[str, Point3D],
+            name: str,
+            dir_: Vector3D,
+        ) -> list[str]:
             if alignments is None:
                 # Use cell_alignment as fallback
-                return [cell_alignment * dir_] * num
+                return [np.array(cell_alignment) * dir_] * num
             if len(alignments) != num:
                 raise ValueError(f"{name}_alignments has a mismatching size.")
-            alignments = list(alignments)
+            alignments_in_list = list(alignments)
             for i in range(num):
-                alignments[i] = mapping[alignments[i]]
-            return alignments
+                alignments_in_list[i] = mapping[alignments_in_list[i]]
+            return alignments_in_list
```

Comment:
Currently, `alignments_in_list` starts as a list of strings and becomes a list of vectors as the `mapping` is applied over each element.

A possible way to avoid this problem cleanly is by replacing all of the selected code by the following:
```py
            alignment_directions = [char_to_direction[char] for char in alignments]
            return alignment_directions
```